Automattic/jetpack
# Charts: Add legend component

## Proposed changes:

* Adds a legend component to the charts library, and adds it to each of the chart types
* The styling (especially the alignment of the legend, as well as positioning of vertical legend) is to come in a future PR

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* In your Jetpack local env, navigate to the Storybook development environment:

   ```
  pnpm i
  cd projects/js-packages/storybook
  pnpm run storybook:dev
  ```

* Find `JS Packages/Charts/Legend` in the sidebar

* Verify the legend
  - Renders correctly with sample data
  - Responds to different data sets via Storybook controls
  - Works with horizontal and vertical configurations
  - Works in each chart option 

|  |  |
| ------------- | ------------- |
| ![image](https://github.com/user-attachments/assets/406c316c-72c6-4584-b6c2-15a46814989c)  | ![image](https://github.com/user-attachments/assets/9642f7ba-5436-44bd-a77b-380af3c65208)  |
| ![image](https://github.com/user-attachments/assets/ae1ca791-ac1a-494e-93e4-b6a92fec28f7)  | ![image](https://github.com/user-attachments/assets/ecfcdfa8-82f7-4087-8b26-2dd398119296)  |
